shared-libs/transitions/test/unit/pregnancy_registration.js

```diff
@@ -401,3 +401,203 @@ describe('pregnancy registration', () => {
+    assert.equal(doc.lmp_date, eightWeeksAgo.clone().toISOString());
+    assert.equal(doc.expected_date, eightWeeksAgo.clone().add(40, 'weeks').toISOString());
+  });
+
+  it('valid adds lmp_date and patient_id', () => {
+    sinon.stub(utils, 'getContactUuid').resolves('uuid');
+    sinon.stub(transitionUtils, 'getUniqueId').resolves(12345);
+
+    const doc = {
+      form: 'l',
+      type: 'data_record',
+      fields: {
+        lmp_date: eightWeeksAgo.valueOf(),
+        patient_name: 'abc'
+      }
+    };
+
+    return transition.onMatch({ doc: doc }).then(function (changed) {
+      assert.equal(changed, true);
+      assert.equal(doc.lmp_date, eightWeeksAgo.toISOString());
+      assert(doc.patient_id);
+      assert.equal(doc.tasks, undefined);
+    });
+  });
+
+  it('valid name invalid LMP date', () => {
+    const doc = {
+      form: 'l',
+      from: '+1234',
+      type: 'data_record',
+      fields: {
+        patient_name: 'hi',
+        lmp_date: 'x'
+      }
+    };
+
+    return transition.onMatch({ doc: doc }).then(function (changed) {
+      assert.equal(changed, true);
+      assert.equal(doc.patient_id, undefined);
+      assert.equal(getMessage(doc),
+        'Date should be later than 40 weeks ago. ' +
+        ' Date should be older than 8 weeks ago.');
+    });
+  });
+
+  it('invalid name invalid LMP Date logic', () => {
+    const doc = {
+      form: 'l',
+      from: '+123',
+      type: 'data_record',
+      fields: {
+        patient_name: '',
+        lmp_date: null
+      }
+    };
+
+    return transition.onMatch({ doc: doc }).then(function (changed) {
+      assert.equal(changed, true);
+      assert.equal(doc.patient_id, undefined);
+      assert.equal(getMessage(doc),
+        'Invalid patient name. ' +
+        ' Date should be later than 40 weeks ago. ' +
+        ' Date should be older than 8 weeks ago.');
+    });
+  });
+  
+  it('LMP date less than 8 weeks ago should fail', () => {
+    const start = moment().subtract({ weeks: 7, days: 6 }).startOf('day');
+    sinon.stub(utils, 'getContactUuid').resolves('uuid');
+    sinon.stub(transitionUtils, 'getUniqueId').resolves(12345);
+
+    const doc = {
+      form: 'l',
+      type: 'data_record',
+      fields: {
+        patient_name: 'abc',
+        lmp_date: start.valueOf()
+      }
+    };
+
+    return transition.onMatch({ doc: doc }).then(function (changed) {
+      assert.equal(changed, true);
+      assert.equal(doc.patient_id, undefined);
+      assert.equal(getMessage(doc), 'Date should be older than 8 weeks ago.');
+    });
+  });
+
+  it('LMP date more than 40 weeks ago should fail', () => {
+    const start = moment().subtract({ weeks: 40, days: 1 }).startOf('day');
+    sinon.stub(utils, 'getContactUuid').resolves('uuid');
+    sinon.stub(transitionUtils, 'getUniqueId').resolves(12345);
+
+    const doc = {
+      form: 'l',
+      type: 'data_record',
+      fields: {
+        lmp_date: start.valueOf(),
+        patient_name: 'abc'
```

Comment:
As above, please add a `reported_date` property to check that validation is triggered with correct data.